android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt

```diff
@@ -378,6 +380,11 @@ open class GestureHandler<ConcreteGestureHandlerT : GestureHandler<ConcreteGestu
     lastAbsolutePositionY = GestureUtils.getLastPointerY(adaptedTransformedEvent, true)
     lastEventOffsetX = adaptedTransformedEvent.rawX - adaptedTransformedEvent.x
     lastEventOffsetY = adaptedTransformedEvent.rawY - adaptedTransformedEvent.y
+
+    if (sourceEvent.action == MotionEvent.ACTION_DOWN || sourceEvent.action == MotionEvent.ACTION_HOVER_ENTER || sourceEvent.action == MotionEvent.ACTION_HOVER_MOVE) {
```

Comment:
(just for clarification) I assume there is no need to set type of pointer on `MotionEvent.ACTION_HOVER_EXIT` right?

Reply:
Yes, I don't think we need to set that on exit. The main idea was to set this variable when gesture receives first event (inside `onPointerDown` on web, `touchesBegan` on iOS, etc.). 

Now I see that in `NativeViewGestureHandler` on iOS i call `setCurrentPointerType` also on `TouchUp` methods. I will check if it is necessary.

Reply:
As I said, I've tested that and it seems that we don't need most of those calls. I removed them in [ab677e2](https://github.com/software-mansion/react-native-gesture-handler/pull/2760/commits/ab677e27049ef754cb2c14a1f4fbdbe63dd18990). I've left one in `handleDragEnter` since I'm not sure when this method is called. If you know how to test it let me know, I'll be grateful, cc @j-piasecki.

Reply:
`handleDragEnter` should be called when pointer moves into the view boundary without touching down. I'm not sure whether it requires to start drag inside the view (touch the view -> drag outside -> drag back inside) or not (touch outside the view -> drag into the view)

Reply:
Okay, I've just checked that and it does require to start drag inside of view. Given that, I removed the line that sets `pointerType` inside that function in [77ab5ce](https://github.com/software-mansion/react-native-gesture-handler/pull/2760/commits/77ab5cecef9433dc6b08c4ca76a17cc9e3f76f7a).